testinfra/utils/ansible_runner.py

```diff
@@ -18,150 +18,189 @@
-            "gather_facts": "no",
-            "tasks": [{
-                "action": action,
-            }],
-        }, variable_manager=self.variable_manager, loader=self.loader)
-        tqm = None
-        callback = Callback()
-        try:
-            tqm = ansible.executor.task_queue_manager.TaskQueueManager(
-                inventory=self.inventory,
-                variable_manager=self.variable_manager,
-                loader=self.loader,
-                options=self.cli.options,
-                passwords=None,
-                stdout_callback=callback,
-            )
-            tqm.run(play)
-        finally:
-            if tqm is not None:
-                tqm.cleanup()
-
-        return callback.result
+        '''Invokes a single module on a single host and returns dict results'''
+
+        # runner must have a directory based payload
+        data_dir = tempfile.mkdtemp()
+        if os.path.exists(data_dir):
+            shutil.rmtree(data_dir)
+        if not os.path.exists(data_dir):
+            os.makedirs(data_dir)
+
+        # runner must have an inventory file
+        inv_dir = os.path.join(data_dir, 'inventory')
+        if not os.path.exists(inv_dir):
+            os.makedirs(inv_dir)
+        inv_file = os.path.join(inv_dir, os.path.basename(self.host_list))
+        shutil.copy(self.host_list, inv_file)
+
+        # molecule inventories use lookups
+        this_env = os.environ.copy()
+        env_keys = list(this_env.keys())
+        for ekey in env_keys:
+            if not ekey.startswith('MOLECULE'):
+                this_env.pop(ekey)
+        env_dir = os.path.join(data_dir, 'env')
+        if not os.path.exists(env_dir):
+            os.makedirs(env_dir)
+        env_file = os.path.join(env_dir, 'envvars')
+        with open(env_file, 'w') as f:
+            f.write('---\n')
+            f.write(yaml.dump(this_env))
+
+        # build the kwarg payload ansible-runner requires
+        runner_kwargs = {
+            'private_data_dir': data_dir,
+            'host_pattern': host,
+            'module': module_name,
+            'module_args': module_args,
+            'json_mode' : True if module_name != "shell" else False
+        }
+
+        # ansible-runner does not have kwargs for these
+        for opt in ['become', 'check']:
+            if kwargs.get(opt, False):
+                if 'cmdline' not in runner_kwargs:
+                    runner_kwargs['cmdline'] = '--%s' % opt
+                else:
+                    runner_kwargs['cmdline'] += ' --%s' % opt
+
+        # invoke ansible-runer -> ansible adhoc
+        r = ansible_runner.run(**runner_kwargs)
+
+        # NOTE(ianw): good heavens, strip ascii color, first line and
+        # newlines from something like:
+        #
+        # ---
+        # localhost | CHANGED | rc=0 >>
+        # Linux
+        #
+        # ---
+        #
+        # I am sure this is not the best way...
+        if module_name == "shell":
```

Comment:
When using the shell module, the output is expected to just be the stdout of whatever was called; strip all the ansible output.  Otherwise, it should return the json output as an object